vispy/visuals/transforms/transform_system.py

```diff
@@ -200,8 +216,8 @@ def configure(self, viewport=None, fbo_size=None, fbo_rect=None,
         # left, whereas framebuffer origin is in bottom left.
         map_from = [(0, 0), canvas.size]
         map_to = [(0, canvas.physical_size[1]), (canvas.physical_size[0], 0)]
-        self._canvas_transform.transforms[1].set_mapping(map_from, map_to)
-
+        self._update_if_maps_changed(self._canvas_transform.transforms[1],
+                                     'ct1', np.array((map_from, map_to)))
```

Comment:
I'm not totally sold on this approach. Are you finding this to be helpful in your performance tests? At least in the ST transform it already does these types of checks: https://github.com/vispy/vispy/blob/master/vispy/visuals/transforms/linear.py#L163. Yes you are repeating some arithmetic in the `set_mapping` method, but overall it still does the checks.

I also don't like the creating of the arrays here or at least we need to be sure things aren't needlessly copied. I know at least the STTransform does a `np.asarray` so this overall array creation should be the net same result and memory copying. Just worried. Lastly, is there maybe a check of `canvas.size`, `canvas.physical_size`, and other size tuples used to short circuit this logic? Or does that make the code too ugly.

Reply:
As I mention above, I am uncertain about this change now lol

Reply:
digging into it again I see a performance hit removing the change I committed - somewhere in the transform system the _set_st is running an update for most visuals due to a scale change (I think its due to a difference between the inbound maps and the z scale from the panzoom camera).

From the debug printouts I have that the difference comes from the scale in z:

[[scale_in, scale_current]
[[ 1.91616766 -9.97058824  1.          1.        ]] [  1.91616762e+00  -9.97058868e+00   9.99999997e-07   1.00000000e+00]
[[translate_in, translate_current]
[[  0.19161677  50.85         0.           0.        ]] [  0.19161677  50.84999847   0.           0.        ]

The values of 1.0 and 9.99999997e-07 differ and the need_update flag is set.

My change bypasses all of this at the scene level (well in the transform system of the scene to be more precise) by never changing the transform to begin with. The camera is changing the scale somewhere else in the widget and the _set_st() call is properly running an update on all the viewbox widgets as the scale is different. It shouldn't be necessary as the code at the picking level didn't change the mapping (based on my picking changes in canvas.py).

I am not sure how to handle the top level scene canvas running a configure() with no real changes and the camera causing (if I am right that it is the source of the scale change) the STTransform changes to propagate everywhere - any thoughts?

The performance change is perceivable in interactions - below is the profiler output for the configure call when I launch the benchmark, maximize window, and pan 4 different viewboxes:

 - removing my change (apply transforms always - I pull transform.set_mapping(new_maps[0], new_maps[1]) back a level of indent in my commit thus forcing it to execute always)
       19    0.000    0.000    1.382    0.073 transform_system.py:178(configure) 
 - apply _update_if_maps_changed as committed based on checking the maps
       19    0.000    0.000    0.343    0.018 transform_system.py:178(configure)